optaplanner-core/src/main/java/org/optaplanner/core/api/score/stream/ConstraintCollectors.java

```diff
@@ -161,6 +163,9 @@
                 () -> new int[1],
                 (resultContainer, a) -> {
                     int value = groupValueMapping.applyAsInt(a);
+                    if (value == 0) {
+                        return NOOP;
```

Comment:
Does this give a noticable performance boost? Is it worth the added cost in code readability?

Reply:
There are two different situations here:

1. The constraint weight is zero, and therefore the rule should not be executed at all. I agree, that should be handled somewhere else. (It isn't handled at all.)
1. The input value to the collector is zero, and therefore there is no change to the collector. I would argue that this is still a valid situation - if you're collecting min/max of something, you can reasonably expect zeros to be there. And in case of a zero, there is no change to the collector - and therefore this is a valid shortcut.

Whether or not this brings actual value in terms of performance, that is hard to say. This PR does so much with performance that I no longer have separate data just for this code change. I can remove this, if you insist.